pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -167,31 +174,147 @@ func checkVersion(instance *servingv1alpha1.KnativeServing) error {
 	return nil
 }
 
-// Check for service mesh memberRoll for maistra
-func verifyServiceMeshMemberRoll(instance *servingv1alpha1.KnativeServing) error {
-	smmr := &maistrav1.ServiceMeshMemberRoll{}
-	checkError := func(err error, instance *servingv1alpha1.KnativeServing) error {
-		msg := fmt.Sprintf("Istio not detected; please install ServiceMesh")
-		instance.Status.MarkDependencyMissing(msg)
-		log.Error(err, msg)
+// createIngressNamespace checks for knative-serving-ingress namespace and creates if doesn't exist
+func createIngressNamespace(instance *servingv1alpha1.KnativeServing) error {
+	ns := &v1.Namespace{}
+	ingressNamespace = instance.GetNamespace() + "-ingress"
```

Comment:
Maybe make this a function `func ingressNamespace(ks *servingv1alpha1.KnativeServing) string`. You can reuse that throughout and avoid the global string which makes me anxious 😂